src/Npgsql/MultiplexingDataSource.cs

```diff
@@ -349,6 +358,20 @@ static void CompleteWrite(NpgsqlConnector connector, ref MultiplexingStats stats
         // ReSharper disable once FunctionNeverReturns
     }
 
+    protected override void DisposeBase()
+    {
+        base.DisposeBase();
+        MultiplexCommandWriter.Complete();
```

Comment:
This means that NpgsqlCommand.Execute will start throwing ChannelClosedException, right? Should we catch that and throw ObjectDisposedException instead?

(should probably add a test exercising that and making sure the right exception is thrown etc.)